safe_transaction_service/events/services/queue_service.py

```diff
@@ -0,0 +1,272 @@
+import json
+import logging
+from typing import Any, Dict, Optional
+
+from django.conf import settings
+
+import pika.exceptions
+from pika import BlockingConnection, URLParameters
+from pika.adapters.gevent_connection import GeventConnection
+from pika.channel import Channel
+from pika.exchange_type import ExchangeType
+
+logger = logging.getLogger(__name__)
+
+
+class QueueServiceProvider:
+    def __new__(cls):
+        if not hasattr(cls, "instance"):
+            if settings.EVENTS_QUEUE_URL_HOST:
+                if settings.EVENTS_QUEUE_ASYNC_CONNECTION:
+                    cls.instance = AsyncQueueService()
+                else:
+                    cls.instance = SyncQueueService()
+            else:
+                # Mock send_event to not configured host us is not mandatory configure a queue for events
+                cls.instance = MockedQueueService()
+        return cls.instance
+
+    @classmethod
+    def del_singleton(cls):
+        if hasattr(cls, "instance"):
+            del cls.instance
+
+
+class QueueService:
+    EXCHANGE_NAME = settings.EVENTS_QUEUE_RABBITMQ_EXCHANGE_NAME
```

Comment:
Set this inside `__init__` (or pass it as a parameter in __init__ and send it on the provider)